include/deal.II/non_matching/mapping_info.h

```diff
@@ -656,9 +656,19 @@ namespace NonMatching
     /**
      * A pointer to the internal data of the underlying mapping.
      */
-    std::shared_ptr<typename Mapping<dim, spacedim>::InternalDataBase>
+    std::unique_ptr<typename Mapping<dim, spacedim>::InternalDataBase>
       internal_mapping_data;
```

Comment:
This effectively deletes the copy constructor and the copy assignment operator. Is that intended? Should we be more explicit about the special member functions?

Reply:
This is a good point. To be honest, I did not think about the copy operations, I merely observed unnecessary copies of `shared_ptr` in the function calls that I wanted to detect, and then I kept the `unique_ptr` because this should not be be private to the present object and not shared with other objects. Of course, we should have a general look at this and the intended use. @bergbauer what is your opinion? Let me mark this as draft as long as this is not resolved.

Reply:
I think you shouldn't copy these objects, `unique_ptr` is the correct choice.

Reply:
If so, could you explicitly `=delete` the copy constructor and copy operator? The errors one gets if they are implicitly deleted are often just difficult to interpret.